src/services/SearchService.php

```diff
@@ -0,0 +1,212 @@
+            case 'events': 
+                // Check we have all required fields
+                if(count(array_diff(['name', 'location', 'description', 'start_date', 'id'], array_keys($data)))) { 
+                    throw new Exception('Missing required fields');
+                }
+
+                $params = [];
+                $params['body']  = [
+                    'title' => $data['name'],
+                    'location' => $data['location'],
+                    'description' => $data['description'],
+                    'stub' => (isset($data['stub']) ? $data['stub'] : ''),
+                    'hashtag' => (isset($data['hashtag']) ? $data['hashtag'] : ''),
+                    'start' => $data['start_date'],
+                    'image' => (isset($data['icon']) ? $data['icon'] : '')
+                ];
+                $params['type']  = 'events';
+                $params['id']    = $data['id'];
+
+                break;
+            case 'talks':
+                // Check we have all required fields
+                if(count(array_diff(['title', 'description', 'id'], array_keys($data)))) { 
+                    throw new Exception('Missing required fields');
+                }
+
+                $params = [];
+                $params['body'] = [
+                    'title' => $data['title'],
+                    'description' => $data['description'],
+                    'speaker' => ((isset($data['speakers']) && is_array($data['speakers'])) ? implode(',', $data['speakers']) : ''),
+                    'start' => isset($data['start_date']) && $data['start_date'] ? $data['start_date'] : null 
+                ];
+
+                $params['type'] = 'talks';
+                $params['id'] = $data['id'];
+
+                break;
+            case 'speakers':
+                // Check we have all required fields
+                if(count(array_diff(['speaker_name', 'speaker_id'], array_keys($data)))) { 
+                    throw new Exception('Missing required fields');
+                }
+
+                $params = [];
+                $params['body'] = [
+                    'name' => $row['speaker_name'],
+                    'start' => null
+                ];
+                $params['type'] = 'speakers';
+                $params['id'] = $row['speaker_id'];
+
+                break;
+
+            default:
+                throw new Exception('Unknown index');
+        }
+
+        $params['index'] = 'ji-search';
+
+        $ret = $this->provider->index($params);
+
+        if(is_array($ret) && $ret['created'] === true) { 
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Remove document from the search index in the provided type
+     * 
+     * @param string $type Define the index's type we want to remove from
+     * @param int $id The ID of the document we want to remove
+     *
+     * @return bool
+     */
+    public function remove($type, $id) {
```

Comment:
It's worth splitting these classes into separate services

This is a very long class as it stands